### PR TITLE
Stop service and remove notification if process stops.

### PR DESCRIPTION
**Pull request by caiofaustino**

### CODE REVIEW

**Comment by caiofaustino** [comment]
> 
> Just a thought, and this doesn't have to be part of this pull request, but if you feel like it, it might also make sense to make sure the UI switch for starting ABCore is set to `false` when running `onDestroy`

yeah I'm planning on doing another PR to work on the UI switch logic a bit

File: app/src/main/java/com/greenaddress/abcore/ABCoreService.java [hunk 1]
```diff
@@ -25,21 +24,154 @@
...
+            }
+        }
+    });
+
+    private final Thread torWaiterThread = new Thread(new Runnable() {
+        @Override
+        public void run() {
+            try {
+                final int exit = mProcessTor.waitFor();
+                Log.i(TAG, "Bitcoin process finished - " + exit);
```
Comment (greenaddress) [inline app/src/main/java/com/greenaddress/abcore/ABCoreService.java]: log line should say Tor process finished

File: app/src/main/java/com/greenaddress/abcore/ABCoreService.java [hunk 2]
```diff
@@ -25,21 +24,154 @@
...
+                stopSelf();
+            } catch (InterruptedException e) {
+                Log.e(TAG, "Bitcoin InterruptedException", e);
```
Comment (greenaddress) [inline app/src/main/java/com/greenaddress/abcore/ABCoreService.java]: this is Tor interrupted

**Comment by caiofaustino** [comment]
Fixed logs and renamed mProcess to mBitcoinProcess for clarity.

**Comment by greenaddress** [comment]
This breaks start on a vanilla install for me - immediately dies, without this it doesn't

**Comment by caiofaustino** [comment]
What dies? The service and the notification?
I think the issue already exists, my chance only make the visual feedback more clear.

 Can you share any logs? I'll investigate more.

**Comment by caiofaustino** [comment]
I'll wait on #106 so that I can do some more testings.

**Comment by caiofaustino** [comment]
I'm closing this until I finish testing other issues I found that relate to this.

**Pull request closed without merging**